src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -54,19 +56,84 @@ namespace OpenMS
       SiriusAdapterAlgorithm();
 
       /// Struct for temporary folder structure
-      struct SiriusTmpStruct
+      struct SiriusTemporaryFileSystemObjects
       {
-        String tmp_dir;
-        String tmp_ms_file;
-        String tmp_out_dir;
-      };
+      public:
 
-      /**
-        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+        SiriusTemporaryFileSystemObjects(int debug_level) : removeable_(false) { constructSiriusTemporaryFileSystemObjects_(debug_level); }
+
+        ~SiriusTemporaryFileSystemObjects(){ if (removeable_) removeSiriusTemporaryFileSystemObjects_(debug_level_, tmp_dir_, tmp_ms_file_); }
+
+        String getTmpDir()
+        {
+          return tmp_dir_;
+        }
+
+        String getTmpOutDir()
+        {
+          return tmp_out_dir_;
+        } 
 
+        String getTmpMsFile()
+        {
+          return tmp_ms_file_;
+        }
+      
+      private:
+        bool removeable_;
+        int debug_level_;
+
+        String tmp_dir_;
+        String tmp_ms_file_;
+        String tmp_out_dir_;
+
+        /**
+        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+        @param debug_level Debug level
         @return SiriusTmpStruct
-      */
-      static SiriusAdapterAlgorithm::SiriusTmpStruct constructSiriusTmpStruct();
+        */
+
+        void constructSiriusTemporaryFileSystemObjects_(int debug_level)
```

Comment:
this function is only called once. Just move its content to the ctor and get rid of it.